tests/test_bookmarks_updated.py

```diff
@@ -0,0 +1,127 @@
+            if keys in extra_stuff:
+                first_sync_bookmark['bookmarks'].pop(keys)
+
+        #######################
+        # Update State between Syncs
+        #######################
+
+        new_state = {'bookmarks': dict()}
+        simulated_states = self.calculated_states_by_stream(first_sync_bookmark)
+
+        for stream, updated_state in simulated_states.items():
+            new_state['bookmarks'][stream] = updated_state
+        menagerie.set_state(conn_id, new_state)
+
+        ###############################
+        # Run Second Sync
+        ###############################
+
+        second_sync_record_count = self.run_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmark = menagerie.get_state(conn_id)
+
+        for stream in testable_streams:
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_replication_method = self.expected_replication_method()
+                expected_replication_keys = self.expected_replication_keys()
+                # information required for assertions from sync 1 and 2 based on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_messages = [record.get('data') for record in first_sync_records.get(stream, {}).get('messages', [])
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in second_sync_records.get(stream, {}).get('messages', [])
+                                        if record.get('action') == 'upsert']
+                first_bookmark_value = first_sync_bookmark.get('bookmarks', {stream: None}).get(stream)
+                first_bookmark_value = list(first_bookmark_value.values())[0]
+                second_bookmark_value = second_sync_bookmark.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_value = list(second_bookmark_value.values())[0]
+
+                replication_key = next(iter(expected_replication_keys[stream]))
+                first_bookmark_value_utc = self.convert_state_to_utc(first_bookmark_value)
+                second_bookmark_value_utc = self.convert_state_to_utc(second_bookmark_value)
+                simulated_bookmark = new_state['bookmarks'][stream]
+                simulated_bookmark_value = list(simulated_bookmark.values())[0]
+
+                # verify the syncs sets a bookmark of the expected form
+                self.assertIsNotNone(first_bookmark_value)
+                self.assertIsNotNone(second_bookmark_value)
+
+                # verify the 2nd bookmark is equal to 1st sync bookmark
+                #TODO : 2nd bookmark value is getting assigned from the execution time rather than the actual bookmark time, check with Dev team on the tap implementation
```

Comment:
This should be documented as a bug or removed from the test depending on discussion with Product/Devs.

Reply:
created a new bug